src/components/Layout/Header/NavBar/NavBar.tsx

```diff
@@ -37,18 +37,17 @@ export const NavBar = (props: NavBarProps) => {
 
   const navItemGroups = useMemo(() => {
     const groups = union(routes, pluginRoutes)
-      .filter((route): route is Extract<Route, { label: string }> =>
-        isLargerThanMd
-          ? !route.disable && !route.hide && !route.hideDesktop
-          : !route.disable && !route.hide && !route.mobileNav,
+      .filter(route => 'label' in route)
+      .filter((route): route is NavRoute =>
+        isLargerThanMd ? !route.disable && !route.hideDesktop : !route.disable && !route.mobileNav,
```

Comment:
This won't work with the new types and would require additional narrowing e.g type guards of sorts.